modules/pricer/src/main/java/com/opengamma/strata/pricer/credit/IsdaCdsProductPricer.java

```diff
@@ -347,6 +347,32 @@ public double riskyAnnuity(
     return riskyAnnuity(cds, rates.getFirst(), rates.getSecond(), referenceDate, stepinDate, effectiveStartDate, priceType);
   }
 
+  /**
+   * Calculates the risky annuity sensitivity of the product.
+   * <p>
+   * The risky annuity sensitivity of the product is the sensitivity of risky annuity to the underlying curves.
+   * The resulting sensitivity is based on the currency of the CDS product.
+   * 
+   * @param cds  the product
+   * @param ratesProvider  the rates provider
+   * @param referenceDate  the reference date
+   * @param refData  the reference data
+   * @return the risky annuity sensitivity
+   */
+  public PointSensitivityBuilder riskyAnnuitySensitivity(
+      ResolvedCds cds,
+      CreditRatesProvider ratesProvider,
+      LocalDate referenceDate,
+      ReferenceData refData) {
+
+    ArgChecker.isTrue(cds.getProtectionEndDate().isAfter(ratesProvider.getValuationDate()), "CDS already expired");
```

Comment:
Would it not be better to return an empty sensitivity? In either case, this case should be in the Javadoc.